core/src/main/proto/io/lionweb/protobuf/Chunk.proto

```diff
@@ -6,54 +6,71 @@ option java_multiple_files = true;
 option java_package = "io.lionweb.protobuf";
 option java_outer_classname = "ChunkProtos";
 
+// We make use of interned strings, languages, and meta-pointers because this is leading to significant performance
+// benefits.
+// So many integer values are indexes into tables of interned values.
+// Values prefixed with "si_" are string indexes
+// Values prefixed with "li_" are language indexes
+// Values prefixed with "mpi_" are meta-pointer indexes
+//
+// Null values for strings and languages are encoded as 0, so indexes into these tables
+// are shifted by one. Specifically:
+// - During serialization: null → 0; value i → index (i + 1), where i is the position in the interned table.
+// - During deserialization: 0 → null; index i → element (i - 1) of the interned table.
+
 message PBChunk {
-  string serializationFormatVersion = 1;
-  // We use this mechanism both to save space and to represent nulls (identified by -1)
-  repeated string stringValues = 2;
-  repeated PBMetaPointer metaPointers= 3;
-  repeated PBLanguage languages = 4;
-  repeated PBNode nodes = 5;
+  string serialization_format_version = 1;
 
+  // Interned strings
+  repeated string interned_strings = 2;
+  repeated PBMetaPointer interned_meta_pointers= 3;
+
+  // Interned languages
+  repeated PBLanguage interned_languages = 4;
+  repeated PBNode nodes = 5;
 }
 
 message PBLanguage {
-  int32 key = 1;
-  int32 version = 2;
+  optional uint32 si_key = 1;
+  optional uint32 si_version = 2;
 }
 
 message PBNode {
-  int32 id = 1;
-  int32 classifier = 2;
+  // Yes, we should _not_ have null IDs, but if necessary we can represent them
+  optional uint32 si_id = 1;
+  uint32 mpi_classifier = 2;
   repeated PBProperty properties = 3;
   repeated PBContainment containments = 4;
   repeated PBReference references = 5;
   /* This is a list of indexes representing the string values corresponding to the IDs of the annotations */
-  repeated int32 annotations = 6;
-  /* Optional*/ int32 parent = 7;
+  /* int to accomodate potential null values */
+  repeated uint32 si_annotations = 6 [packed = true];
```

Comment:
I think in [proto3](https://protobuf.dev/programming-guides/proto3/#use-packed) `packed=true` is the default, so we can omit it?

Reply:
Yes, it is true, however it may be not obvious, so I would prefer to have it and I added it also below for children